packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -527,6 +527,32 @@ export class ProfileInfo {
         return new Session(sessCfg);
     }
 
+    /**
+     * Checks if a profile (including nested subprofiles) exists in the given profiles object.
+     * The profile name can use dot notation for nested profiles (e.g., "main.sub1.sub2").
+     *
+     * @param profileName - The dotted path to the profile (e.g., "main.sub1.sub2")
+     * @param profilesObj - The profiles object to search within
+     * @returns true if the profile exists, false otherwise
+     */
+    private static profileExists(profileName: string, profilesObj: any): boolean {
```

Comment:
Can we come up with a more explicit type for `profilesObj`? Generally its discouraged to explicitly type a parameter as `any` unless we cannot determine the type at the time of the function call.

Reply:
😅 yes sorry